hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -1809,7 +1809,17 @@ protected void secondPassCompileForeignKeys(
 					secondPassCompileForeignKeys( referencedClass.getSuperclass().getTable(), done, buildingContext );
 				}
 
-				fk.setReferencedTable( referencedClass.getTable() );
+				if ( referencedClass.isAbstract() && referencedClass.getSubclassSpan() == 1 ) {
+					Iterator<PersistentClass> iter = referencedClass.getSubclassClosureIterator();
```

Comment:
referencedClass.isAbstract() may be null so if( referencedClass.isAbstract() ... can cause a NPE
I would change the code to 
if ( referencedClass.isAbstract() != null && referencedClass.isAbstract() && referencedClass.getSubclassSpan() == 1 ) {